pilopt/src/lib.rs

```diff
@@ -432,6 +434,31 @@ fn extract_constant_lookups<T: FieldElement>(pil_file: &mut Analyzed<T>) {
     }
 }
 
+/// Extracts constraints from lookups where the LHS' selector = 0.
+fn extract_empty_permutations<T: FieldElement>(pil_file: &mut Analyzed<T>) {
```

Comment:
In which way does it extract? It simplifies, doesn't it?

Reply:
To be accurate, it transforms. I guess we can call it a simplification

Reply:
I would expect a function called `extract_...` to return something - the stuff it extracted.